layers/+lang/go/funcs.el

```diff
@@ -37,7 +37,9 @@
 (defun spacemacs//go-setup-backend-lsp ()
   "Setup lsp backend"
   (if (configuration-layer/layer-used-p 'lsp)
-      (lsp)
+      (progn
+        (setq-local lsp-prefer-flymake :none)
```

Comment:
This one should not be hardcoded since if someone wants to use lsp-ui it wont be possble.

Reply:
@yyoncho I think my latest commit should resolve your concern. Provide the ability to set the option as a layer variable, and have its value default to a safe value for the Go layer (`:none`).